### PR TITLE
Fix cache invalidate test

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###
Fix cache invalidate test.
cache invalite test sometime fails as  it seems there is sync. issue with threads.
hope cachecleanup triggers maintaining the cache.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by mdayican** [comment]
> Why not revert the commit https://build-beta.platform.hmcts.net/job/HMCTS_BSP/job/bulk-scan-orchestrator/job/master/49/changes and then fix the test and functionality together ?

as we do not use it, we do not need to revert whole thing. this is just to leave the master in peace.

**Comment by nitinprabhu** [comment]
> > Why not revert the commit https://build-beta.platform.hmcts.net/job/HMCTS_BSP/job/bulk-scan-orchestrator/job/master/49/changes and then fix the test and functionality together ?
> 
> as we do not use it, we do not need to revert whole thing. this is just to leave the master in peace.

Removing test will make master green but we will end up something in master which is not working consistently. So I would suggest even if it is not getting used let's remove it and merge with tests once it's fixed :)

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java [hunk 1]
```diff
@@ -250,7 +251,7 @@ public void should_invalidate_userDetail_when_cached_token_removed_from_cache()
         assertThat(userDetailsBefore).isEqualTo(expectedUserDetails1);
 
         idamCachedClient.removeAccessTokenFromCache(jurisdiction1);
-        idamCachedClient.cleanUpAccessTokenCache();
+        idamCachedClient.cleanUpCaches();
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:254]: I realise it was like that before, but does this test verify the behaviour of `removeAccessTokenFromCache` if we clean the whole cache afterwards?

Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:254]: it is not cleaning the cache from javadoc of method:
  /**
   * Performs any pending maintenance operations needed by the cache. Exactly which activities are
   * performed -- if any -- is implementation-dependent.
   */

Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:254]: first i tried 3 sec sleep, it passed test but it was luck.

Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:254]: Not sure we need `cleanUpCaches` to exist at all

Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:254]: aaa, I see

**Comment by mdayican** [comment]
> Suggestion for a future PR: ideally, get rid of the public cleanup method, so that the cache is easier to use. If not, change its name, because cleanup often means remove everything from.

tidyUpCache?
maintainCache?

**Comment by lgonczar** [comment]
> tidyUpCache?
> maintainCache?
> I can fix now, as I am doing some commits to trigger builds to see it always pass.

`forceCacheUpdate()`?

**Comment by mdayican** [comment]
open 
https://github.com/hmcts/bulk-scan-orchestrator/pull/945

**Pull request closed without merging**